test/e2e/framework/util.go

```diff
@@ -5205,5 +5205,8 @@ func GetClusterZones(c clientset.Interface) (sets.String, error) {
 			zones.Insert(zone)
 		}
 	}
+	if zones.Len() == 0 {
+		return nil, fmt.Errorf("zone name for node %v not found. No label with key %s", nodes, kubeletapis.LabelZoneFailureDomain)
```

Comment:
Sorry I didn't catch this earlier: The original `getZoneNames()` fails as soon as a node is missing zone label. We should do the same thing here, i.e. move it up and have it as the `else` clause inside the for loop.

Reply:
I put get node err check before GetClusterZones, return err when some nodes are missing zone label.